retail/interactive-tutorial/RetailProducts.Samples/ImportProductsGcsSample.cs

```diff
@@ -0,0 +1,130 @@
+﻿// Copyright 2022 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START retail_import_products_from_gcs]
+// Import products into a catalog from GCS using Retail API
+
+using Google.Cloud.Retail.V2;
+using Google.LongRunning;
+using System;
+
+/// <summary>
+/// The import products GCS sample class.
+/// </summary>
+public class ImportProductsGcsSample
+{
+    /// <summary>
+    /// Get import products GCS request.
+    /// </summary>
+    /// <param name="gcsObjectName">The name of the GCS object.</param>
+    /// <param name="projectId">The current project id.</param>
+    /// <returns>The import products request.</returns>
+    public static ImportProductsRequest GetImportProductsGcsRequest(string gcsObjectName, string projectId, string productsBucketName)
+    {
+        string gcsBucket = $"gs://{productsBucketName}";
+        string gcsErrorsPrefix = $"{gcsBucket}/error";
+
+        string locationId = "global";
+        string catalogId = "default_catalog";
+        string branchId = "default_branch";
+        // To check error handling paste the invalid catalog name here:
+        // catalogId = "invalid_catalog_name";
+        BranchName defaultBranch = new BranchName(projectId, locationId, catalogId, branchId);
+
+        var gcsSource = new GcsSource();
+        gcsSource.InputUris.Add($"{gcsBucket}/{gcsObjectName}");
```

Comment:
Collection initializer here as well.

Reply:
We can't, `InputUris` readonly property.

Reply:
See my comment on #1817:

When you do `InputUris = { "uri" }`, you are not setting `InputUris`, internally that's just calling the `Add` method. What you can't do is `InputUris = new List<string> { "uri" }` because that would first be setting `InputUris` to a new list and then calling the `Add` method.

Please change here and everywhere.